sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java

```diff
@@ -2533,6 +2596,58 @@
             });
     }
 
+    /**
+     * Check that the additional headers field is present and add the additional auth header
+     *
+     * @param headerName name of the header to be added
+     * @param context current request context
+     *
+     * @return boolean representing the outcome of adding header operation
+     */
+    private void addSupplementaryAuthHeader(String headerName, String entity, Context context) {
+        context.getData(AZURE_REQUEST_HTTP_HEADERS_KEY)
+            .map(headers -> {
+                if (headers instanceof HttpHeaders) {
+                    HttpHeaders customHttpHeaders = (HttpHeaders) headers;
+                    customHttpHeaders.add(headerName, entity);
+                }
+                return null;
+            });
+    }
+
+    /**
+     * Checks if the given entity is an absolute URL, if so return it.
+     * Otherwise, construct the URL from the given entity and return that.
+     *
+     * @param entity : entity to forward messages to.
+     *
+     * @return Forward to Entity represented as an absolute URL
+     */
+    private String getAbsoluteUrlFromEntity(String entity) {
+        // Check if passed entity is an absolute URL
+        try {
+            URL url = new URL(entity);
+            return url.toString();
+        } catch (MalformedURLException ex) {
+            // Entity is not a URL, continue.
+        }
+        UrlBuilder urlBuilder = new UrlBuilder();
+        urlBuilder.setScheme("https");
+        urlBuilder.setHost(managementClient.getEndpoint());
+        urlBuilder.setPath(entity);
+
+        try {
+            URL url = urlBuilder.toUrl();
+            return url.toString();
+        } catch (MalformedURLException ex) {
+            // This is not expected.
+            logger.error("Failed to construct URL using the endpoint:'{}' and entity:'{}'",
```

Comment:
Would this be a fatal error? Do we want to silently continue if we were unable to create a URL instead of returning null? Could the user recover from this?

Reply:
yes, the request will fail with 400. so it better for us to fail from here directly.